test/tools/utils.js

```diff
@@ -0,0 +1,291 @@
+ * @param {String} dataFile saved data file path
+ * @param {{min: Number, max: Number}} dataRange
+ * @return {Object}
+ */
+function prepareInputsData(inputsDataInfo, dataFile, dataRange) {
+  const dstDataDict = {inputsData: {}};
+  let srcDataDict = {};
+  if (fs.existsSync(dataFile)) {
+    srcDataDict = readJsonFile(dataFile);
+  }
+  for (const source in inputsDataInfo) {
+    // reserve last input data when generating new required input data
+    if (srcDataDict['inputsData'] !== undefined &&
+        srcDataDict['inputsData'][source] !== undefined) {
+      dstDataDict['inputsData'][source] = srcDataDict['inputsData'][source];
+    } else {
+      const targetDataInfo = inputsDataInfo[source];
+      if (targetDataInfo.data !== undefined) {
+        const srcDataInfo = inputsDataInfo[targetDataInfo.data];
+        const inputTensor = new Tensor(
+            srcDataInfo.shape, dstDataDict['inputsData'][targetDataInfo.data]);
+        let outputTensor;
+        if (targetDataInfo.processCategory === 'transpose') {
+          outputTensor = transpose(
+              inputTensor, {permutation: targetDataInfo.permutation});
+        } else if (targetDataInfo.processCategory === 'negative') {
+          outputTensor = neg(inputTensor);
+        }
+        dstDataDict['inputsData'][source] = outputTensor.data;
+      } else {
+        const total = sizeOfShape(targetDataInfo.shape);
+        if (targetDataInfo.dataRange !== undefined) {
+          // Specified data range
+          dataRange = targetDataInfo.dataRange;
+        }
+        const generatedNumbers = getRandomNumbers(total, dataRange, targetDataInfo.dataType);
+        dstDataDict['inputsData'][source] = generatedNumbers;
+      }
+    }
+  }
+  return dstDataDict;
+}
+
+/**
+ * Get JSON object from specified JSON file.
+ * @param {String} filePath
+ * @return {Object}
+ */
+function readJsonFile(filePath) {
+  let inputFile;
+  if (path.isAbsolute(filePath)) {
+    inputFile = filePath;
+  } else {
+    inputFile =
+        path.join(path.dirname(process.argv[1]), filePath);
+  }
+  const content = fs.readFileSync(inputFile).toString();
+  const jsonDict = JSON.parse(
+      content.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, // remove comments
+          (m, g) => g ? '' : m));
+  return jsonDict;
+}
+
+/**
+ * Save JSON infomation into file.
+ * @param {Object} jsonDict
+ * @param {String} saveFile
+ */
+function writeJsonFile(jsonDict, saveFile) {
+  const parentDirectory = path.dirname(saveFile);
+  if (!fs.existsSync(parentDirectory)) {
+    fs.mkdirSync(parentDirectory);
+  }
+  const jsonString = JSON.stringify(jsonDict, function(key, value) {
+    // the replacer function is looking for some typed arrays.
+    // If found, it replaces it by a trio
+    if ( value instanceof Int8Array ||
+         value instanceof Uint8Array ||
+         value instanceof Int32Array ||
+         value instanceof Uint32Array ||
```

Comment:
Should `Uint16Array` be included in here for the float16 cases? (ignore this comment if not relevant, or if Node.js now ships with `Float16Array` on by default)

Reply:
Now we can import **Float16Array** from [@petamoriken/float16](https://www.npmjs.com/package/@petamoriken/float16), so here we can remove workaround `Uint16Array`.